New serverless pattern - API Gateway - Lambda - Redshift Data API

klmuthu opened the pull request:
*Issue #, if available:*

*Description of changes:*

Patter to expose data from Redshift through API Gateway using Lambda and Redshift Data API.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

ellisms commented:
Hi @klmuthu I'm facing a dependency error when installing requirements. Can you verify the aws-cdk-lib version?
Error: `ERROR: No matching distribution found for aws-cdk-lib==2.118.0`

ellisms commented:
Hi @klmuthu I think you are missing the cdk.json file. When deploying, I receive this error:
`--app is required either in command-line, in cdk.json or in ~/.cdk.json`

Also, please include a completed example-pattern.json file

klmuthu commented:
Hi @ellisms I updated the readme to use --app with cdk deploy. Also, updated the readme to create python virtual environment to avoid version conflicts. I didnt create cdk.json just to keep the repo small. Let me know if this deploys  now. Please note the Redshift cluster has to be created as a prerequisite and the .env should be updated with the arn. The instructions is in the readme.

ellisms commented:
Thanks @klmuthu . I'm able to deploy the solution, but when testing, I'm getting the following error in the Lambda function:
`[ERROR] Runtime.ImportModuleError: Unable to import module 'redshiftapi': No module named 'redshiftapi'`

klmuthu commented:
Hi @ellisms, Thanks for testing. There was a copy/paste error. I fixed and tested it. Please try again.

ellisms commented:
Thanks. The function now executes, but returns this error:
`"body": "{\"error\": \"FATAL: database \\\"sample-data-dev\\\" does not exist\"}"`
The database does exist and I can connect via the Console.

klmuthu commented:
Hey @ellisms, I believe it could be because the db name not correct. You used hyphen instead of underscore in the db name. The name should be sample_data_dev. Can you give this a try?

ellisms commented:
> Hey @ellisms, I believe it could be because the db name not correct. You used hyphen instead of underscore in the db name. The name should be sample_data_dev. Can you give this a try?

Good catch! Tested and it is working. Left a few feedback items for review/updates.

klmuthu commented on apigw-lambda-redshiftdataapi/README.md:
````diff
@@ -0,0 +1,57 @@
+1. Install dependencies:
+   ```
+   pipenv install
+   pipenv requirements > requirements.txt
````
done

klmuthu commented on apigw-lambda-redshiftdataapi/README.md:
```diff
@@ -0,0 +1,67 @@
+- [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+- [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+- [AWS CDK installed](https://docs.aws.amazon.com/cdk/latest/guide/cli.html)
+- [Python 3 installed](https://www.python.org/downloads/)
```
bumped cdk version to 2.172, deployed and tested.

klmuthu commented on apigw-lambda-redshiftdataapi/README.md:
```diff
@@ -0,0 +1,67 @@
+
+## Testing
+
+1. Use an api client with oauth capability such as Postman or Rapid API to test the api. API url can be found from CDK stack deployment output. You will need to login to AWS Console, navigate to Cognito app client and retrieve the client id and secret to pass it in the api call.
```
done

klmuthu commented:
Hi @ellisms, make the changes you requested.

ellisms commented:
Thanks @klmuthu . I'll add this to the publishing queue.
